cram/cram_codecs.c

```diff
@@ -1416,25 +1410,36 @@ cram_codec *cram_byte_array_len_decode_init(char *data, int size,
     c->decode = cram_byte_array_len_decode;
     c->free   = cram_byte_array_len_decode_free;
     
-    cp += itf8_get(cp, &encoding);
-    cp += itf8_get(cp, &sub_size);
+    cp += safe_itf8_get(cp, endp, &encoding);
+    cp += safe_itf8_get(cp, endp, &sub_size);
+    if (sub_size < 0 || endp - cp < sub_size)
```

Comment:
Should this be sub_size < 0 || encoding < 0 and setting both of these to -1 instead of 0 above?

Right now it looks like a failure to decode isn't caught here, instead accepted 0,0 as the defaults . This doesn't sound right.